src/test/java/com/wojciechkocik/usage/service/CourseUsageServiceTest.java

```diff
@@ -0,0 +1,217 @@
+package com.wojciechkocik.usage.service;
+
+import com.wojciechkocik.usage.dto.CourseUsageCreate;
+import com.wojciechkocik.usage.dto.DailyUsageResponse;
+import com.wojciechkocik.usage.entity.CourseUsage;
+import com.wojciechkocik.usage.repository.CourseUsageRepository;
+import org.jfairy.Fairy;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import javax.validation.ConstraintViolationException;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+/**
+ * @author Wojciech Kocik
+ * @since 22.03.2017
+ */
+@RunWith(SpringRunner.class)
+@SuppressWarnings("unused")
+@SpringBootTest
+public class CourseUsageServiceTest {
```

Comment:
Overall summary of tests:
1. Integration tests are good, but we we would like to see unit tests also, at least for TimeSpentMidnightCrossService.
2. When you write test they should follow the same rules as prodaction code: they should be compact and easy to read, and without duplication.
3. Setup-phases obscure tests, and it is hard to see what conditions the test validates.